tools/packaging/scripts/configure-hypervisor.sh

```diff
@@ -48,13 +48,27 @@ recognised_tags=(
 	[speed]="maximise startup speed"
 )
 
+# Wrapper around the bc command.
+# Return 0 if bc expression $1 is true, otherwise 1.
+calc() {
+	local expr="$1"
+	[ "$(echo $expr | bc)" == "1" ]
+}
+
 # Display message to stderr and exit indicating script failed.
 die() {
 	local msg="$*"
 	echo >&2 "$script_name: ERROR: $msg"
 	exit 1
 }
 
+# Compare with the QEMU version.
+# Use a bc expression as $1.
+qemu_version_is() {
+	local expr="$1"
+	calc "${qemu_version} $expr"
```

Comment:
`qemu_version` is local in `main`, how can you access it?, also afaik `bc` is unable to compare versions for exaple:

```sh
$ echo "5.0.0 > 2.7" | bc
(standard_in) 1: syntax error
```

Reply:
Yep. I think you need to pass two version number strings and then either:

- Ensure both version numbers are in `major.minor` format and use `bc(1)`.
- Use @devimc's trick of using `sort -V` to handle semver version numbers if they are in `major.minor.patch`.

Reply:
I wonder in fact whether it would make sense to write a `kata-version-compare.sh` script as it feels similar in some ways to https://github.com/kata-containers/tests/blob/main/.ci/kata-arch.sh (ideally with a set of bats unit tests ;)

However, I'm wondering if there is a rust/golang cli program that can do all this which we can standarise on (*surely* there is by now?!? :smile:)

Reply:
https://github.com/timvisee/version-compare looks interesting, but doesn't provide a simple CLI which is odd :confused:.

Reply:
Hmmm.  Parsing version numbers is nearly always a bad idea.  It very easily breaks if there are downstream versions of the package that have different features backported.

If at all possible, it's better to instead probe for the particular features that are required in the new version, rather than examining the version number itself.

Reply:
@devimc , @jodh-intel , @dgibson , I appreciated your comments but I'd like to propose that the changes to `configure-hypervisor.sh` gets merged as is; then I open another issue for improving the script. is it reasonable?

Reply:
@wainersm - regardless @dgibson 's comment (good catch!), I think that your implementation of `qemu_version_is` is weak and buggy, isn't it? :thinking:

Reply:
@devimc I just realize I was testing the script with a VERSION file were I put only "x.y". I fixed the script to correctly parse "x.y.z", ie., the `qemu_version` will be two digits "x.y". IMHO it is not needed to consider "z" on that script because it indicates the release of a stable branch (which in turn are security fixes only) which is unlikely to introduce new configure options.

Reply:
hmm... right... `calc` is still buggy but since you will work on the changes proposed by @dgibson 

lgtm

Reply:
@devimc you mean buggy because passing a 3-digits version to the function then `bc` is gonna fail?

Reply:
yes

```
+ calc '2.0.0 >= 5'
+ local 'expr=2.0.0 >= 5'
++ echo 2.0.0 '>=' 5
++ bc
(standard_in) 1: syntax error
+ '[' '' == 1 ']'
```